pkg/endpointmanager/host.go

```diff
@@ -47,14 +47,14 @@ func (mgr *EndpointManager) OnUpdateNode(oldNode, newNode *v1.Node,
 	oldNodeLabels := oldNode.GetLabels()
 	newNodeLabels := newNode.GetLabels()
 
+	node.SetLabels(newNodeLabels)
```

Comment:
I moved this above GetHostEndpoint() as in tests we didn't have the host endpoint and thus wouldn't have set the labels. With this PR calling `SetLabels` will actually inform the subscribers synchronously, and in this case nodediscovery to sync the labels to CiliumNode. This also is why we can remove the `ciliumNodeUpdater` from `watchers/node.go`.